File: LinkedList/1D LinkedList/searching/searching.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) { this.val = val; }
}

class Solution {
    public boolean searchKey(ListNode head, int key) {
        ListNode temp = head;
        while (temp != null) {
            if (temp.val == key) {
                return true;
            }
            temp = temp.next;
        }
        return false;
    }
}

File: LinkedList/1D LinkedList/searching/searching.ts
class ListNode {
    val: number;
    next: ListNode | null;

    constructor(val: number, next: ListNode | null = null) {
        this.val = val;
        this.next = next;
    }
}

class Solution {
    searchKey(head: ListNode | null, key: number): boolean {
        let temp = head;
        while (temp !== null) {
            if (temp.val === key) {
                return true;
            }
            temp = temp.next;
        }
        return false;
    }
}
